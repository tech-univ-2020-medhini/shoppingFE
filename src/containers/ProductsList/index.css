.products-list{
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    padding: 1rem;
    justify-content: space-around;
}
.categories-text{
    display:flex;
    padding-left: 2rem;
    padding-top: 2rem;
    font-size: xx-large;
}