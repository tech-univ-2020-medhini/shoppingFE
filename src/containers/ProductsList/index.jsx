import React from 'react';
import PropTypes from 'prop-types';
import ProductCard from '../../components/ProductCard';
import FilterBar from '../../components/FilterBar';
import './index.css';

const ProductList = (props) => {
  const { filterCategory, productList } = props;
  let products;
  if (filterCategory === 'All') {
    productList.forEach((product) => {
      products.push(<ProductCard
        id={product.id}
        name={product.name}
        price={product.price}
        cart={product.cart}
        imageLink={product.imageLink}
        key={product.id}
      />);
    });
  } else if (filterCategory !== undefined) {
    products = productList.map((product) => (
      <ProductCard
        id={product.id}
        name={product.name}
        price={product.price}
        cart={product.cart}
        imageLink={product.imageLink}
        key={product.id}
      />
    ));
  }


  return (
    <div className="Song-page">
      <p>
        {filterCategory}
      </p>
      <FilterBar />
      <div className="songs-container">
        {products}
      </div>
    </div>
  );
};
ProductList.propTypes = {
  filterCategory: PropTypes.string.isRequired,
  productList: PropTypes.arrayOf(PropTypes.object).isRequired,
};
export default ProductList;
