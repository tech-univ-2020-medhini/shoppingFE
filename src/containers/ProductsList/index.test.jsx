import React from 'react';
import { render } from '@testing-library/react';
import ProductsList from './index';


describe('the products page container', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<ProductsList />);

    expect(asFragment()).toMatchSnapshot();
  });
});
