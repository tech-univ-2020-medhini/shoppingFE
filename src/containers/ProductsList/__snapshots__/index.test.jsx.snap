// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the products page container should render correctly 1`] = `
<DocumentFragment>
  <div
    class="Song-page"
  >
    <p />
    <div
      class="filter-bar"
    >
      <button
        class="category-button"
        type="button"
      >
        All
      </button>
    </div>
    <div
      class="songs-container"
    />
  </div>
</DocumentFragment>
`;
